BridgeAppSDK/ORKResult+SBAExtension.swift

```diff
@@ -0,0 +1,341 @@
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import ResearchKit
+
+private var kStartDateKey = "startDate"
+private var kEndDateKey = "endDate"
+private var kIdentifierKey = "identifier"
+
+private var kItemKey = "item"
+
+private var kTappingViewSizeKey = "TappingViewSize"
+private var kButtonRectLeftKey = "ButtonRectLeft"
+private var kButtonRectRightKey = "ButtonRectRight"
+private var kTapTimeStampKey = "TapTimeStamp"
+private var kTapCoordinateKey = "TapCoordinate"
+private var kTappedButtonIdKey = "TappedButtonId"
+private var kTappedButtonNoneKey = "TappedButtonNone"
+private var kTappedButtonLeftKey = "TappedButtonLeft"
+private var kTappedButtonRightKey = "TappedButtonRight"
+private var kTappingSamplesKey = "TappingSamples"
+
+private var kSpatialSpanMemoryGameStatusKey = "MemoryGameStatus"
+private var kSpatialSpanMemoryGameStatusUnknownKey = "MemoryGameStatusUnknown"
+private var kSpatialSpanMemoryGameStatusSuccessKey = "MemoryGameStatusSuccess"
+private var kSpatialSpanMemoryGameStatusFailureKey = "MemoryGameStatusFailure"
+private var kSpatialSpanMemoryGameStatusTimeoutKey = "MemoryGameStatusTimeout"
+
+private var kSpatialSpanMemorySummaryNumberOfGamesKey = "MemoryGameNumberOfGames"
+private var kSpatialSpanMemorySummaryNumberOfFailuresKey = "MemoryGameNumberOfFailures"
+private var kSpatialSpanMemorySummaryOverallScoreKey = "MemoryGameOverallScore"
+private var kSpatialSpanMemorySummaryGameRecordsKey = "MemoryGameGameRecords"
+
+private var kSpatialSpanMemoryGameRecordSeedKey = "MemoryGameRecordSeed"
+private var kSpatialSpanMemoryGameRecordGameSizeKey = "MemoryGameRecordGameSize"
+private var kSpatialSpanMemoryGameRecordGameScoreKey = "MemoryGameRecordGameScore"
+private var kSpatialSpanMemoryGameRecordSequenceKey = "MemoryGameRecordSequence"
+private var kSpatialSpanMemoryGameRecordTouchSamplesKey = "MemoryGameRecordTouchSamples"
+private var kSpatialSpanMemoryGameRecordTargetRectsKey = "MemoryGameRecordTargetRects"
+
+private var kSpatialSpanMemoryTouchSampleTimeStampKey = "MemoryGameTouchSampleTimestamp"
+private var kSpatialSpanMemoryTouchSampleTargetIndexKey = "MemoryGameTouchSampleTargetIndex"
+private var kSpatialSpanMemoryTouchSampleLocationKey = "MemoryGameTouchSampleLocation"
+private var kSpatialSpanMemoryTouchSampleIsCorrectKey = "MemoryGameTouchSampleIsCorrect"
+
+private var QuestionResultQuestionTextKey = "questionText"
+private var QuestionResultQuestionTypeKey = "questionType"
+private var QuestionResultUserInfoKey = "userInfo"
+
+enum ResultType: Int {
+    case URL, Dictionary, Data
+}
+
+class ArchiveableResult : NSObject {
```

Comment:
Would be cleaner to use a struct here with the result, resultType and filename defined as let. Then it is a clearly immutable object that is used to pass some bits of information.